CoolCurrency/Controller/HomeCurrencyViewController.swift

```diff
@@ -106,5 +113,31 @@ extension HomeCurrencyViewController: ViewModelDelegate {
         self.currencyTableView.reloadData()
         self.currencyPickerView.reloadAllComponents()
         self.activityLoader.stopAnimating()
+        self.sendMessage()
+    }
+}
+
+// MARK: - Watch Session Functions
+extension HomeCurrencyViewController: WCSessionDelegate {
+    
+    func session(_ session: WCSession, activationDidCompleteWith activationState: WCSessionActivationState, error: Error?) { }
+    func sessionDidBecomeInactive(_ session: WCSession) { }
+    func sessionDidDeactivate(_ session: WCSession) { }
+    
+    private func sendMessage() {
+        watchSession?.sendMessage(viewModel.currencyDataModelForWatchApp() ?? ["Not Set": ["greyArrow", "Not Set"]],
+                                  replyHandler: nil,
+                                  errorHandler: nil)
+    }
+    
+    func session(_ session: WCSession, didReceiveMessage message: [String: Any]) {
+        DispatchQueue.main.async {
+            if let value = message["getExchangeRate"] as? String {
+                self.viewModel.selectedCurrency = value
+                self.viewModel.fetchCurrencyListFromDatabase(for: self.viewModel.convertCurrencyToCode(for: self.viewModel.retrieveSelectedCurrency))
```

Comment:
Hmm this function doesn't read very well, see if you can alter how they work so that you can just call `fetchCurrencyListFromDatabase()` Too many functions nested in functions here.

Reply:
To make things cleaner as well create a let here that contains the result from `self.viewModel.retrieveSelectedCurrency`, you use it here multiple times so storing it quickly and just using the variable would be cleaner